doc/icache.rst

```diff
@@ -149,3 +167,53 @@ Cache invalidation
 After reset, and when requested by the core (due to a FENCE.I instruction), the whole cache is invalidated.
 Requests are inserted to invalidate the tag RAM for all ways in each cache line in sequence.
 While the invalidation is in-progress, lookups and instruction fetches can proceed, but nothing will be allocated to the cache.
+
+Detailed behaviour
+^^^^^^^^^^^^^^^^^^
+
+This section describes the expected behaviour of the cache, in order to allow functional verification.
+This isn't an attempt to describe the cache's performance characteristics.
+
+The I$ has a single clock (``clk_i``) and asynchronous reset (``rst_ni``).
+
+Data is requested from the instruction memory with the ports prefixed by ``instr_``. These work as described in :ref:`instruction-fetch`.
+Note that there's one extra port on the I$, which doesn't appear at the ``ibex_core`` top-level.
+This is ``instr_pmp_err_i``.
+If the PMP block disallows a fetch for a certain address, it will squash the outgoing memory request entirely and set ``instr_pmp_err_i``.
+If that happens, the cache behaves as if the request was granted but returned some time later with an error.
+Note that it is possible for ``instr_gnt_i`` and ``instr_pmp_err_i`` to be high on the same cycle (if the memory system ties ``instr_gnt_i`` to ``instr_req_o``, for example).
+In that case, the error signal takes precedence.
+
+Fetched instructions are returned to the core using ports ``ready_i``, ``valid_o``, ``rdata_o``, ``addr_o`` and ``err_o``.
+This interface uses ready/valid handshaking in the usual way (a transaction is signalled by ready and valid being high; if ready or valid goes high, it will remain high until the transaction goes through; if valid goes high, the other output signals will remain stable until the transaction goes through).
```

Comment:
This is sort-of true, but the stability won't hold if there is an intervening branch.

Reply:
Ok, that makes sense. (I also made a mistake about the ready signal that I've fixed).